qiskit_nature/drivers/molecule.py

```diff
@@ -396,3 +396,49 @@ def perturbations(self) -> Optional[List[float]]:
     def perturbations(self, value: Optional[List[float]]) -> None:
         """Set perturbations"""
         self._perturbations = value
+
+    @property
+    def core_orbitals(self) -> List[int]:
```

Comment:
I simple copied this over from `QMolecule`. Tbh I did not give it much more thought but I guess what the logic entails is to consider all but the highest shell to be "core". Then again, this also makes an assumption about the used basis set so I am really not sure what originally went into this.

Reply:
Another comment (since I just redirected @woodsp-ibm) here: I agree, that we can think about moving this elsewhere. Below are the `grep`-results for `core_orbitals` on the Qiskit Nature code base:
```sh
% grep -R 'core_orbitals' qiskit_nature                                                                                                                                                                                                        10:37:03
qiskit_nature/drivers/molecule.py:    def core_orbitals(self) -> List[int]:
qiskit_nature/drivers/qmolecule.py:    def core_orbitals(self) -> List[int]:
qiskit_nature/drivers/qmolecule.py:            logger.info("Core orbitals list %s", self.core_orbitals)
qiskit_nature/drivers/second_quantization/qmolecule.py:    def core_orbitals(self) -> List[int]:
qiskit_nature/drivers/second_quantization/qmolecule.py:            logger.info("Core orbitals list %s", self.core_orbitals)
qiskit_nature/transformers/second_quantization/freeze_core_transformer.py:            1. When `freeze_core` is enabled (the default), the `core_orbitals` listed in the
qiskit_nature/transformers/second_quantization/freeze_core_transformer.py:                        `QMolecule.core_orbitals`.
qiskit_nature/transformers/second_quantization/electronic/freeze_core_transformer.py:            1. When `freeze_core` is enabled (the default), the `core_orbitals` listed in the
qiskit_nature/transformers/second_quantization/electronic/freeze_core_transformer.py:                        `Molecule.core_orbitals`.
qiskit_nature/transformers/second_quantization/electronic/freeze_core_transformer.py:        inactive_orbs_idxs = molecule.core_orbitals
```
As you can see, it is indeed only being used by the `FreezeCoreTransformer`. However, since it is part of the public API I am not too sure if we can move it entirely away from our `molecule`-representation object as people may depend on it.
However, Panos' above comment is also valid: Is the current way even general enough for this approach to be valid once we go to basis sets larger than `sto3g`? Also: what if a user already employed an ECP in his calculation? Then the number of core orbitals will also be wrong.

Reply:
Why do you think it is wrong - it computes the number of occupied molecular orbitals that comprise the core. We have done computations with other basis sets for smaller molecules.

If you use ECP then yes it will go wrong and you should not be using freeze core with that. Given how ECP is setup for  driver etc we had no easy way to detect if ECP was used but needed to rely on the user at that level.

Reply:
> For each atom it figures the nearest noble gas and sums over those particles

I am not fully sure I understand this. Isn't it simply doing the following:
- iterate all atoms
- if in second row, take one MO as core (equivalent to arguing the `1s` AO can be set to core)
- if in third row, take 4 MOs as core (`2s, 2px, 2py, 2pz`) 
- and so on...

Even if we were to go to a large basis (say e.g. `ccpV5z` which has 201 orbitals for water), this approach would still be valid because the larger basis does not change the core because it only introduces additional orbitals for higher shells, correct @pbark?

I agree, that the ECP case should be handled by the user.

With that said, should we move this logic to the `FreezeCoreTransformer`? Or do we need it available more generally (i.e. on `Molecule`?)

Reply:
By row you mean row of the periodic table - since each row ends with the noble gas that is why I indicated that since they are the Z numbers listed there and it uses the nearest lowest one for that number of electrons from it. At least thats how I think of it.

ECP and the orbitals do need handling. It does not even result is a reduction of the number of qubits - at least for G16 when I looked it adds back in a equivalent number of orbitals, with high occupancy energy, which I guess one can remove, but by default we do not save qubits hence we had no way of easily knowing etc.

For me this really is not a function of Molecule since that was more about representing a molecule like the xyz format where its distances, atoms etc.  It could be a util - or since its needed by FreezeCoreTransformer put it there and allow its access as a public function should it be needed elsewhere. Of course there one would have to pass in a list of atoms - I am not sure in the context of how we use it how doable that is - before it used to be on the driver result (QMolecule) - maybe your driver result now always has a Molecule and hence why you want to out it there,

Reply:
I see. Yes, a `Molecule` will now always be present. I just noticed though, that the `Molecule` has no notion of orbitals so I agree that we should extract this to the transformer (at least for the time being). Will also update this tomorrow.